dagger/hilt-node-factory-compiler/src/main/kotlin/com/bumble/appyx/dagger/hilt/node/AppyxNodeFactoryProcessor.kt

```diff
@@ -0,0 +1,127 @@
+package com.bumble.appyx.dagger.hilt.node
+
+import com.google.auto.common.GeneratedAnnotations
+import com.google.auto.service.AutoService
+import com.squareup.javapoet.*
+import net.ltgt.gradle.incap.IncrementalAnnotationProcessor
+import net.ltgt.gradle.incap.IncrementalAnnotationProcessorType.ISOLATING
+import javax.annotation.processing.AbstractProcessor
+import javax.annotation.processing.Processor
+import javax.annotation.processing.RoundEnvironment
+import javax.lang.model.SourceVersion
+import javax.lang.model.element.Modifier
+import javax.lang.model.element.TypeElement
+import javax.lang.model.util.Elements
+
+/**
+ * Source generator to support Hilt injection of AppyxNodeFactories.
+ *
+ * Should generate:
+ * ```
+ * @Module
+ * @InstallIn(SingletonComponent.class)
+ * @OriginatingElement(topLevelClass = $.class)
+ * public interface $_NodeFactoryModule {
+ *   @Binds
+ *   @IntoMap
+ *   @ClassKey(FactoryClass.class)
+ *   Object binds(FactoryImplClass impl)
+ * }
+ * ```
+ */
+@AutoService(Processor::class)
+@IncrementalAnnotationProcessor(ISOLATING)
+class AppyxNodeFactoryProcessor : AbstractProcessor() {
+
+    override fun getSupportedAnnotationTypes() =
+        setOf(ClassNames.NODE_FACTORY_ANNOTATION.canonicalName())
+
+    override fun getSupportedSourceVersion(): SourceVersion = SourceVersion.latest()
+
+    override fun process(
+        annotations: MutableSet<out TypeElement>,
+        roundEnv: RoundEnvironment
+    ): Boolean {
+        annotations.firstOrNull()?.also { processingAnnotation ->
+            roundEnv.getElementsAnnotatedWith(processingAnnotation).forEach {
+                processElement(it as TypeElement)
+            }
+        }
+        return false
+    }
+
+    private fun processElement(typeElement: TypeElement) {
+        val bindingClassName = ClassName.get(typeElement)
+        val superTypeClassName = ClassName.get(typeElement.interfaces[0])
```

Comment:
This is very very brittle. I will make finding the interface safer if we want to go ahead with this change